libs/ui-lib/lib/ocm/components/clusterConfiguration/ExternalPlatformDropdown.tsx

```diff
@@ -8,107 +8,100 @@ import {
-        style={{ float: 'right' }}
-      >
-        Learn more <i className="fas fa-external-link-alt" />
-      </a>
-    </DropdownItem>,
-    <DropdownItem key="nutanix-platform" id="nutanix">
-      {externalPlatformTypes['nutanix']}
-      <a
-        href={NUTANIX_CONFIG_LINK}
-        target="_blank"
-        rel="noopener noreferrer"
-        style={{ float: 'right' }}
-      >
-        {'Learn more'} <i className="fas fa-external-link-alt" />
-      </a>
-    </DropdownItem>,
-    <DropdownItem key="vsphere-platform" id="vsphere">
-      {externalPlatformTypes['vsphere']}
-      <a
-        href={VSPHERE_CONFIG_LINK}
-        target="_blank"
-        rel="noopener noreferrer"
-        style={{ float: 'right' }}
-      >
-        {'Learn more'} <i className="fas fa-external-link-alt" />
-      </a>
-    </DropdownItem>,
-  ];
-
-  if (isOracleCloudPlatformIntegrationEnabled) {
-    enabledItems.push(
-      <DropdownItem
-        key="oracle-platform"
-        id="oci"
-        tooltip={isOciDisabled ? disabledOciTooltipContent : undefined}
-        isAriaDisabled={!!isOciDisabled}
-      >
-        {externalPlatformTypes['oci']}
-        <a
-          href={VSPHERE_CONFIG_LINK}
-          target="_blank"
-          rel="noopener noreferrer"
-          style={{ float: 'right' }}
-        >
-          {'Learn more'} <i className="fas fa-external-link-alt" />
-        </a>
-      </DropdownItem>,
-    );
-  }
+  const enabledItems = Object.keys(externalPlatformTypes)
+    .filter((platformType) => {
+      if (platformType === 'oci') {
+        return isOracleCloudPlatformIntegrationEnabled;
+      }
+      return true;
+    })
+    .map((platformType) => {
+      const { label, href, tooltip } = externalPlatformTypes[platformType as ExternalPlatformType];
+      const isOracleDisabled = platformType === 'oci' && isOciDisabled;
+      const isHrefEmpty = href === '';
+      return (
+        <>
+          <DropdownItem
+            key={platformType}
+            id={platformType}
+            tooltip={isOracleDisabled ? disabledOciTooltipContent : tooltip}
+            isAriaDisabled={isOracleDisabled}
+          >
+            {label}
+            {!isHrefEmpty && (
+              <a href={href} target="_blank" rel="noopener noreferrer" style={{ float: 'right' }}>
+                Learn more <i className="fas fa-external-link-alt" />
+              </a>
+            )}
+          </DropdownItem>
+        </>
+      );
+    });
 
   const onSelect = React.useCallback(
     (event?: React.SyntheticEvent<HTMLDivElement>) => {
-      const selectedPlatform = event?.currentTarget.id as PlatformType;
+      const selectedPlatform = event?.currentTarget.id as ExternalPlatformType;
       setValue(selectedPlatform);
       setOpen(false);
-      setCurrent(externalPlatformTypes[selectedPlatform]);
+      setCurrentPlatform(externalPlatformTypes[selectedPlatform].label);
+      onChange(selectedPlatform === 'oci');
```

Comment:
I think this is confusing. It's best that you pass the `selectedPlatform` and then apply the needed logic in the `onChange` handler.